### PR TITLE
XS-4212

### PR DESCRIPTION
**Pull request by allenbross-wf**
### Changes:

- Initial implementation of DQC.US.0011
- Incremented version numbers

### Pull request process

- [ ] Full description of changes provided above.
- [ ] Reviewed by another person. (+1)
- [ ] If this is a code change, reviewed by a second person. (+1)
- [ ] QA'd to the specifications listed [here](https://github.com/DataQualityCommittee/dqc_us_rules#quality-assurance-qa-of-a-pull-request). Documentation only changes do not require the formal code quality assurance process. (+10)

### CODE REVIEW

**Comment by joyreistad-wf** [comment]
+1

File: dqc_us_rules/dqc_us_0011.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    :rtype: None
+    """
+    model_xbrl = val.modelXbrl
+    for check in _load_checks(model_xbrl):
+        rule_index_key = check.rule_num
+        try:  # allow exceptions when no fact or concept for QName
+            for n_fact in model_xbrl.factsByQname[check.nondim_concept]:
+                # here want dimensionless line items only
+                if not n_fact.context.qnameDims:
+                    # find fact expressed with dimensions
```
Comment (derekgengenbacher-wf) [inline dqc_us_rules/dqc_us_0011.py]: Perhaps we could do a list comprehension here to remove a layer of indentation. 

n_facts = [fact for fact in model_xbrl.factsByQname[check.nondim_concept] if not n_fact.context.qnameDims]

File: dqc_us_rules/dqc_us_0011.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                    for d_fact in model_xbrl.factsByQname[check.dim_concept]:
+                        if (_check_for_exclusions(d_fact)):
```
Comment (derekgengenbacher-wf) [inline dqc_us_rules/dqc_us_0011.py]: We could probably do another list comprehension here as well:

d_fact = [fact for fact in model_xbrl.factsByQname[check.dim_concept] if (_check_for_exclusions(fact)]

**Comment by derekgengenbacher-wf** [comment]
+1

**Comment by joyreistad-wf** [comment]
+1

**Comment by joyreistad-wf** [comment]
#### QA Steps:

- Manual testing: Ran the the attached filing through Arelle using this version of DQC as a plugin and saw the following DQC.US.0011 errors
`
[DQC.US.0011.6820] It is expected that Accumulated Other Comprehensive Income (Loss), Net of Tax with a value of -100 should be equal to the dimensionally qualified concept, 
Stockholders' Equity, Including Portion Attributable to Noncontrolling Interest with the dimension us-gaap:StatementEquityComponentsAxis = us-gaap:AccumulatedOtherComprehensiveIncomeMember and a reported value of Stockholders' Equity, Including Portion Attributable to Noncontrolling Interest multiplied by 1. 
This error occurs when the dimensionally qualified fact is not equal to the value input for the standalone item. - abn-20170425.xml 28, 29
[DQC.US.0011.6823] It is expected that Accumulated Other Comprehensive Income (Loss), Net of Tax with a value of -100 should be equal to the dimensionally qualified concept, 
Stockholders' Equity, Including Portion Attributable to Noncontrolling Interest with the dimension us-gaap:StatementEquityComponentsAxis = us-gaap:AccumulatedOtherComprehensiveIncomeMember and a reported value of Stockholders' Equity, Including Portion Attributable to Noncontrolling Interest multiplied by 1. 
This error occurs when the dimensionally qualified fact is not equal to the value input for the standalone item. - abn-20170425.xml 28, 29
`
- Flake8 on dqc_us_rules:
```
(dqc)  joyreistad [dqc] ((HEAD detached at allen/XS-4212)) $ flake8 dqc_us_rules/
(dqc)  joyreistad [dqc] ((HEAD detached at allen/XS-4212)) $
```
- Flake8 on tests:
```
(dqc)  joyreistad [dqc] ((HEAD detached at allen/XS-4212)) $ flake8 tests
(dqc)  joyreistad [dqc] ((HEAD detached at allen/XS-4212)) $
```
- Nosetest result:
```
(dqc)  joyreistad [dqc] ((HEAD detached at allen/XS-4212)) $ nosetests tests
.............................................................................................................
----------------------------------------------------------------------
Ran 109 tests in 0.129s

OK
(dqc)  joyreistad [dqc] ((HEAD detached at allen/XS-4212)) $
```

#### Result: QA +10

File: dqc_us_rules/dqc_us_0011.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    for fact_axis, fact_dim_value in fact.context.segDimValues.items():
+        mem_name = fact_dim_value.memberQname.localName
+        axis_name = fact_axis.qname.localName
+        if not fact_dim_value.isTyped and \
+                ('LegalEntityAxis' == axis_name and
+                    'ScenarioPreviouslyReportedMember' == mem_name or
+                    'StatementScenarioAxis' == axis_name and
+                    'RestatementAdjustmentMember' == mem_name or
+                    'StatementScenarioAxis' == axis_name and
+                    'ScenarioPreviouslyReportedMember' == mem_name):
```
Comment (chrislococo-wf) [inline dqc_us_rules/dqc_us_0011.py]: I would pull this whole if out into a helper method for readability/documentation

**Comment by chrislococo-wf** [comment]
Doh, too late. Possible cleanups in the future.

**Pull request merged**